src/entry-client.js

```diff
@@ -79,21 +86,21 @@ export default function initializeClient(createApp, clientOpts) {
 
         // After routing, unregister any dynamic Vuex modules from prior components
         router.afterEach((to, from) => {
-            const priorComponents = router.getMatchedComponents(from);
-
-            priorComponents
+            const fetchDataArgs = { app, route: to, router, store, from };
+            const toModuleNames = router.getMatchedComponents(to).map(c => getModuleName(c, to));
+            router.getMatchedComponents(from)
                 .filter(c => 'vuex' in c)
+                .filter(c => !shouldIgnoreRouteUpdate(c, fetchDataArgs))
                 .forEach((c) => {
                     const fromModuleName = getModuleName(c, from);
-                    const toModuleName = getModuleName(c, to);
```

Comment:
This was a bug before - we needed to get the `to` module name from the proper components matched against the `to` route